Classes/Auth/TMOAuth.m

```diff
@@ -75,6 +90,37 @@ - (id)initWithURL:(NSURL *)URL method:(NSString *)method postParameters:(NSDicti
 
 #pragma mark - Private
 
+NSMutableDictionary *generateOAuthParameters(TMOAuth *tmOAuth,
+                                             NSURL *URL,
+                                             NSString *method,
+                                             NSDictionary *postParameters,
+                                             NSString *nonce,
+                                             NSString *consumerKey,
+                                             NSString *consumerSecret,
+                                             NSString *token,
+                                             NSString *tokenSecret,
+                                             NSString *timestamp) {
+    NSMutableDictionary *oAuthParameters = [[NSMutableDictionary alloc] initWithDictionary:@{
+                                                                                              @"oauth_timestamp" : timestamp,
+                                                                                              @"oauth_nonce" : nonce,
+                                                                                              @"oauth_version" : @"1.0",
+                                                                                              @"oauth_signature_method" : @"HMAC-SHA1",
+                                                                                              @"oauth_consumer_key" : consumerKey,
+                                                                                              }];
+    if (token && token.length > 0)
+        oAuthParameters[@"oauth_token"] = token;
+
+    NSDictionary *queryParameters = TMQueryStringToDictionary(URL.query);
+    NSString *baseURLString = [[URL absoluteString] componentsSeparatedByString:@"?"][0];
+    NSString *baseString = generateBaseString(baseURLString, method, oAuthParameters, queryParameters, postParameters);
+    oAuthParameters[@"oauth_signature"] = sign(baseString, consumerSecret, tokenSecret);
+
+    if (tmOAuth != nil)
```

Comment:
I think it makes more sense if we move the `_baseString` setting stuff back into `initWithURL:method:postParameters:nonce:consumerKey:consumerSecret:token:tokenSecret:` and omit the `tmOAuth` parameter in here altogether. Setting `_baseString` feels more like an `init` thing to me, and having this method modify an object that you pass in seems kinda weird and side-effect-y to me.